tskit_arg_visualizer/__init__.py

```diff
@@ -304,61 +304,71 @@ def _convert_edges_table(ts, recombination_nodes_to_merge):
+        # iterate over unique parent/child combos. Take advantage of the fact that edges
+        # in a tree sequence are always ordered by parent ID.
+        for parent, edges in itertools.groupby(ts.edges(), operator.attrgetter("parent")):
+            parent_links = []  # all links for this parent
             if parent in recombination_nodes_to_merge:
-                parents[i] -= 1
-
-        uniq_child_parent = np.unique(np.column_stack((ts.edges_child, parents)), axis=0) #Find unique parent-child pairs.
-        links = []
-        for ID, combo in enumerate(uniq_child_parent):
-            child = combo[0]
-            parent = combo[1]
-            equivalent_edges = ts.tables.edges[np.where((ts.edges_child == child) & (parents == parent))[0]]
-            region_size = 0
-            bounds = ""
-            for edge in equivalent_edges:
-                bounds += f"{edge.left}-{edge.right} "
-                region_size += edge.right - edge.left
-            alternative_child = ""
-            alternative_parent = ""
-            if ts.tables.nodes.flags[parent] != 131072:
-                children = np.unique(ts.tables.edges[np.where(parents == parent)[0]].child)
-                if len(children) > 2:
-                    alternative_child = children[np.where(children != child)][0]
-                elif len(children) > 1:
-                    alternative_child = children[np.where(children != child)][0]
+                parent -= 1
+                ID = links[-1][0]["id"]  # to account for prev parent_links being overwritten
+            else:
+                edges_for_child = {}  # This is a new parent: make a new array
+            for edge in edges:
+                if edge.child not in edges_for_child:
+                    edges_for_child[edge.child] = [edge]
                 else:
-                    alternative_child = -1 # this occurs when converting from SLiM simulations, needs to have better handling
-                if alternative_child in recombination_nodes_to_merge:
-                    alternative_child -= 1
-            elif parent in recombination_nodes_to_merge:
-                parent = edge.parent - 1
-            if ts.tables.nodes.flags[child] == 131072:
+                    edges_for_child[edge.child].append(edge)
+            for child, equivalent_edges in edges_for_child.items():
+                region_size = 0
+                bounds = ""
+                for edge in equivalent_edges:
+                    bounds += f"{edge.left}-{edge.right} "
+                    region_size += edge.right - edge.left
+                alternative_child = ""
+                alternative_parent = ""
+                if (ts.nodes_flags[parent] & msprime.NODE_IS_RE_EVENT) != 0:
+                    children = np.array(list(edges_for_child.keys()))
+                    if len(children) > 2:
+                        alternative_child = children[np.where(children != child)][0]
```

Comment:
Copied from the original code, but this and the line below give equivalent `alternative_child` definitions, so I don't understand why the  `if len(children) > 2` clause is needed.

Reply:
Correct that `len(children) > 2` clause is not needed at the moment. It was from an initial attempt at polytomies where the JavaScript needs to consider multiple other edges to determine pathing rather than just one other edge. Haven't had time to work on that so it defaults to the same as `len(children) > 1`.